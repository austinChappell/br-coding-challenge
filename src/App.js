import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
} from 'react-router-dom';
import { Provider } from 'react-redux';

// redux store
import store from './data/redux/store';

// pages and layout components
import BaseLayout from './BaseLayout';
import Error404 from './pages/Error404';
import Restaurants from './pages/Restaurants';

const App = () => (
  <div className="App">
    <Provider store={store}>
      <Router>
        <BaseLayout>
          <Switch>
            <Route exact path="/" component={Restaurants} />
            <Route path="/*" component={Error404} />
          </Switch>
        </BaseLayout>
      </Router>
    </Provider>
  </div>
);

export default App;
