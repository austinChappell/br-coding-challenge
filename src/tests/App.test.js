import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json';

import App from '../App';

configure({ adapter: new Adapter() });

describe('The App Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<App />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
