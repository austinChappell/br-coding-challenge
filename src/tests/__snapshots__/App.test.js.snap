// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App Component should render correctly 1`] = `
<div
  className="App"
>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <BrowserRouter>
      <withRouter(Connect(BaseLayout))>
        <Switch>
          <Route
            component={[Function]}
            exact={true}
            path="/"
          />
          <Route
            component={[Function]}
            path="/map"
          />
          <Route
            component={[Function]}
            path="/*"
          />
        </Switch>
      </withRouter(Connect(BaseLayout))>
    </BrowserRouter>
  </Provider>
</div>
`;
